CONTRIBUTING.md

```diff
@@ -1,219 +1,193 @@
+
+> NOTE: Set an environment variable `JAVA_HOME` to the path name of the directory into which you installed the JDK release.
+
+**Prerequisite 2** - Node (v8.9.x or latest LTS release) + npm (v5.6.0 or later)
+
+**Prerequisite 3** - Docker 
+
+#### Building from the source
+
+Clone this repository using the following command.
+
+```
+git clone --recursive https://github.com/ballerinalang/ballerina
+```
+
+If you download the sources, you need to update the git submodules using the following command.
+
+```
+git submodule update --init 
+```
+
+Run the gradle command `./gradlew clean build` from the Ballerina root directory:
+
+Extract the Ballerina distribution created at `distribution/zip/ballerina/target/ballerina-<version>-SNAPSHOT.zip`. The `zip/ballerina` directory contains the runtime only. `zip/ballerina-tools/` contains the runtime and tools (e.g., Ballerina Composer).
+
+### Setting up your development environment
+
+#### Setup IntelliJ IDEA
+
+##### Importing a Ballerina Project
+
+Import the Ballerina project to IntelliJ IDEA similar to any other gradle project. 
+
+* Navigate to `File | Open` menu.
+* Browse the filesystem and Select Ballerina Project's root directory.
+* In "Import Project from Maven" Wizard, set followings.
+   * If prompted to import gradle project select `Auto import`.
+   * Set Project SDK as Java 1.8
+
+##### Useful IDEA Plugins
+
+* ANTLR 4 - https://plugins.jetbrains.com/plugin/7358-antlr-v4-grammar-plugin 
+* Ballerina - https://plugins.jetbrains.com/plugin/9520-ballerina (Need IDEA 2016.3 or newer)
+
+#### Setup Eclipse
+
+##### Importing a Ballerina Project
+
+All Ballerina repositories are developed as maven or gradle projects. So you can import any Ballerina project to Eclipse similar to any maven or gradle project. 
+
+* Navigate to **File** → **Import..**
+* Select **Existing Maven Projects** under **Maven**
+* Next, browse the file system, and open the Maven module.
+
+##### Installing ANTLR4 plugin
+If you are working with the grammar, it would be useful to have ANTLR 4 IDE plugin installed for Eclipse. To install this using Eclipse marketplace:
+
+1. Navigate to **Help** → **Eclipse Marketplace**
+
+2. Search for **antlr4**
+
+3. Install ANTLR 4 IDE plugin (which supports antlr 4.x version).
+
+To open the plugin views: 
+
+1. Navigate to **Window** → **Show View** → **Other**. 
+
+2. Under **ANTLR4** section, select and enable **Parse Tree** view and **Syntax Diagram** views.
+
+### Working With Ballerina Grammar
+
+Ballerina grammar has been implemented using ANTLR4. It is recommended to have some basic understanding on anltr4 grammar syntax and concepts, before working with the Ballerina grammar. “Parr, Terence (January 15, 2013), The Definitive ANTLR 4 Reference” is a good reference to get started with.
+
+Ballerina grammar can be found under docs/grammar in the Ballerina repository. It consists of two files:
```

Comment:
The grammar is at `<BALLERINA_REPO_ROOT>/compiler/ballerina-lang/src/main/resources/grammar`

Reply:
Shall we link github url here?